### ISSUE TITLE
endpoint security api calls for elasticsearch and kibana diagnostics:

### ISSUE THREAD

**Issue by 111andre111** [elastic/support-diagnostics#512]
[1,2,2,1]

Due to the upcoming version 7.14.0, support diagnostics should now obtain following api entries:
Kibana:
```
Endpoint Metadata            : /api/endpoint/metadata/{agent.id}
Endpoint Policy              : /api/fleet/package_policies/{Endpoint.policy.applied.id}
Endpoint Policy response     : /api/endpoint/policy_response?agentId={agent.id}
Security packages data       : /api/fleet/epm/packages?experimental=true&category=security
Trusted apps                 : /api/endpoint/trusted_apps
Associated Fleet Agent status: /api/fleet/agents/{elastic.agent.id}
Agent status 2               : /api/endpoint/action_status?agent_ids={elastic.agent.id}
Transforms                   : /api/transform/transforms
Transform status/logs for the endpoint.metadata_current transform: /api/transform/transforms/{transform.name}/messages  transform.name e.g. endpoint.metadata_current-default-0.20.2
Transform stats logs         : /api/transform/transforms/_stats
Associated Fleet Agent Logs  : /api/fleet/agent_policies/{agent.id} und .item.package_policies   and then  /api/fleet/epm/packages/{package.name}  e.g. system-1.0.0   or --> They can also be access via Logs app with the following filter: elastic_agent.id:{elastic.agent.id} and (data_stream.dataset:elastic_agent)
```
Elasticsearch (I am not sure but maybe this is already obtained by the Elasticsearch diagnostics).
```
/_transform/_stats
```

**Comment by cybersecdiva** [comment]
@spong and I also had this same discussion on getting detection APIs back in August for diagnostics: https://github.com/elastic/support-diagnostics/issues/479#issuecomment-897019236

**Comment by PhaedrusTheGreek** [comment]
It seems that in order to collect this type of data, we must set up CLI flags for the diagnostic tool, and then ask customers to call the diagnostic with the specific agent ID on the command line, which hopefully doesn't have to happen too often.

Given that there's no simple automation potential, and that the case percentage is rare, it may not belong in the capture tool.   

What about if we had a small script that would grab this data when devs ask for it?

**Comment by cybersecdiva** [comment]
@PhaedrusTheGreek  That is a good suggestion and solution and we can definitely implement this action and test it out to see if this would be viable. 😄

**Comment by pickypg** [comment]
I would go a little further and suggest that you all work with the Kibana devs to implement APIs that do not require fetching by ID and instead can fetch all of them, similar to other stat / info collection APIs.

**Comment by 111andre111** [comment]
@pickypg @PhaedrusTheGreek 
Here are meanwhile the fetch all API calls investigated:

```
Endpoint Metadata            : /api/endpoint/metadata       paginated and pages can be increased via pageSize=x
Endpoint Policy              : /api/fleet/package_policies  paginated and pages can be increased via perPage=x
Associated Fleet Agent status: /api/fleet/agents            paginated and pages can be increased via perPage=x
Security packages data       : /api/fleet/epm/packages?experimental=true&category=security
Trusted apps                 : /api/endpoint/trusted_apps   paginated and pages can be increased via per_page=x
                               /api/exception_lists/items/_find?list_id=endpoint_trusted_apps&namespace_type=agnostic&page=1&per_page=10&sort_field=name&sort_order=asc
Event Filters                : /api/exception_lists/items/_find?list_id=endpoint_event_filters&namespace_type=agnostic&page=1&per_page=10&sort_field=created_at&sort_order=desc
Host Isolation Exceptions    : /api/exception_lists/items/_find?list_id=endpoint_host_isolation_exceptions&namespace_type=agnostic&page=1&per_page=10&sort_field=created_at&sort_order=desc
Security Exception List      : /api/exception_lists/_find?namespace_type=agnostic
Transforms                   : /api/transform/transforms
Transform stats logs         : /api/transform/transforms/_stats
```
Some of them are paginated, but I think we already have a mechanism for paginating results. I think we already do the same for the alerts API in Kibana diagnostics already.

**Comment by stefnestor** [comment]
FWIW Transforms & its stats return in ES diag [here](https://github.com/elastic/support-diagnostics/blob/main/src/main/resources/elastic-rest.yml#L483-L493). I can take first stab at PR from https://github.com/elastic/support-diagnostics/issues/512#issuecomment-1022723390 if you want? Do you know the first available versions per endpoint? Are they all 7.14.0?

**Comment by 111andre111** [comment]
https://github.com/elastic/support-diagnostics/pull/603
